File: project_euler/Problem033/Problem033.c
#include <stdio.h>

static int common_divisor(int i, int j);

void Problem033(void) {
    int numerator = 1;
    int denominator = 1;
    for (int i = 1; i < 10; ++i) {
        for (int j = i + 1; j < 10; ++j) {
            if (!((9 * i * j) % (10 * i - j)) && (9 * i * j) / (10 * i - j) < 10) {
                numerator *= i;
                denominator *= j;
            }
        }
    }
    printf("%d\n", denominator / common_divisor(denominator, numerator));
}

static int common_divisor(int i, int j) {
    while (i % j) {
        int auxiliary = i % j;
        i = j;
        j = auxiliary;
    }
    return j;
}

File: project_euler/Problem033/Problem033.py
class Problem033:
    def solution(self) -> None:
        numerator, denominator = 1, 1
        for i in range(1, 10):
            for j in range(i + 1, 10):
                if not (9 * i * j) % (10 * i - j) and (9 * i * j) // (10 * i - j) < 10:
                    numerator *= i
                    denominator *= j
        print(denominator // self.__common_divisor(denominator, numerator))

    def __common_divisor(self, i: int, j: int) -> int:
        while i % j:
            i, j = j, i % j
        return j
